q2_qemistree/_prune_hierarchy.py

```diff
@@ -0,0 +1,73 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2018, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import pandas as pd
+from skbio import TreeNode
+
+
+def prune_hierarchy(feature_data: pd.DataFrame,
+                    tree: TreeNode,
+                    prune_type: str = 'classyfire',
+                    classyfire_level: str = 'class') -> TreeNode:
+    '''This function prunes the tree for visualization. It retains only
+    the features that have been annotated for a user-specified `prune_type`
+
+    Paramters
+    ---------
+    feature_data : pd.DataFrame
+        Feature data table with Classyfire annotations and/or SMILES. Output
+        of make_hierarchy() or get_classyfire_taxonomy()
+    tree : skbio.TreeNode
+        Tree of relatedness of molecules. Output of make_hierarchy()
+    prune_type : str, default 'classyfire'
+        Metadata category for tree pruning ('smiles' or 'classyfire')
+    classyfire_level : str, default 'class'
+        One of the Classyfire levels in ['kingdom', 'superclass', 'class',
+        'subclass', 'direct_parent']
```

Comment:
Can you elaborate on what's the rationale for having two different categories? Seems to me like a single argument would be enough. This argument would take any of the following values: `['kingdom', 'superclass', 'class', 'subclass', 'direct_parent', 'smiles']`.